src/Psalm/Internal/Analyzer/Statements/Expression/AssertionFinder.php

```diff
@@ -1249,9 +1249,16 @@ protected static function processCustomAssertion(
                         $if_types[$var_id] = [[$assertion->rule[0][0]]];
                     }
                 } elseif (\is_string($assertion->var_id)
-                    && $expr instanceof PhpParser\Node\Expr\MethodCall
+                    && (
+                        $expr instanceof PhpParser\Node\Expr\MethodCall
+                        || $expr instanceof PhpParser\Node\Expr\StaticCall
+                    )
                 ) {
-                    $if_types[$assertion->var_id] = [[$assertion->rule[0][0]]];
+                    $var_id = $assertion->var_id;
+                    if (strpos($var_id, 'self::') === 0) {
+                        $var_id = $this_class_name . '::' . substr($var_id, 6);
+                    }
+                    $if_types[$var_id] = [[$assertion->rule[0][0]]];
```

Comment:
This feels hacky, is there a better way?

Reply:
Which bit feels hacky?

Reply:
String operations seems out of place at that point. Also I'm not sure if replacing `self` with `$this_class_name` is 100% correct.

Reply:
@muglug is there anything I can do to make this PR merge-worthy?

Reply:
No, this is fine.